plugins/es-discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -50,68 +52,55 @@ class AwsEc2ServiceImpl implements AwsEc2Service {
-        final AmazonEC2 client = new AmazonEC2Client(credentials, configuration);
-        return client;
-    }
 
-    // pkg private for tests
-    static ClientConfiguration buildConfiguration(Ec2ClientSettings clientSettings) {
-        final ClientConfiguration clientConfiguration = new ClientConfiguration();
+        // NOT supported by AWS SDK V2
         // the response metadata cache is only there for diagnostics purposes,
         // but can force objects from every response to the old generation.
-        clientConfiguration.setResponseMetadataCacheSize(0);
-        clientConfiguration.setProtocol(clientSettings.protocol);
-        if (Strings.hasText(clientSettings.proxyHost)) {
-            // TODO: remove this leniency, these settings should exist together and be validated
-            clientConfiguration.setProxyHost(clientSettings.proxyHost);
-            clientConfiguration.setProxyPort(clientSettings.proxyPort);
-            clientConfiguration.setProxyUsername(clientSettings.proxyUsername);
-            clientConfiguration.setProxyPassword(clientSettings.proxyPassword);
+        // clientConfiguration.setResponseMetadataCacheSize(0);
+
+        ApacheHttpClient.Builder httpClientBuilder = ApacheHttpClient.builder()
+            .socketTimeout(Duration.ofMillis(clientSettings.readTimeoutMillis))
+            .proxyConfiguration(proxyConfig.build());
+
+        ClientOverrideConfiguration.Builder overrideConfig = ClientOverrideConfiguration.builder();
+        overrideConfig.retryPolicy(RetryPolicy.builder(RetryMode.STANDARD).build());
+
+        var clientBuilder = Ec2Client.builder()
+            .credentialsProvider(buildCredentials(clientSettings))
+            .httpClientBuilder(httpClientBuilder)
+            .overrideConfiguration(overrideConfig.build());
+
+        if (Strings.hasText(clientSettings.endpoint)) {
+            LOGGER.debug("using explicit ec2 endpoint [{}]", clientSettings.endpoint);
+            clientBuilder.endpointOverride(URI.create(clientSettings.endpoint));
+        } else {
+            LOGGER.debug("using ec2 region [{}]", Region.AWS_GLOBAL);
+            Ec2Client.builder().region(Region.AWS_GLOBAL);
         }
-        // Increase the number of retries in case of 5xx API responses
-        final Random rand = Randomness.get();
-        final RetryPolicy retryPolicy = new RetryPolicy(
-            RetryPolicy.RetryCondition.NO_RETRY_CONDITION,
-            (originalRequest, exception, retriesAttempted) -> {
-                // with 10 retries the max delay time is 320s/320000ms (10 * 2^5 * 1 * 1000)
-                LOGGER.warn("EC2 API request failed, retry again. Reason was:", exception);
-                return 1000L * (long) (10d * Math.pow(2, retriesAttempted / 2.0d) * (1.0d + rand.nextDouble()));
-            },
-            10,
-            false);
-        clientConfiguration.setRetryPolicy(retryPolicy);
```

Comment:
Couldn't find `v2` equivalent but there are different back off strategies (`FullJitterBackoffStrategy`, `FixedDelayBackoffStrategy`, or `EqualJitterBackoffStrategy`) we can use instead?

Reply:
Looks like still available? https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/RetryPolicy.html